File: eye_stream.py

import cv2


cap = cv2.VideoCapture("rtsp://192.168.71.50:8554/live/eyes")

while True:
    if cap.isOpened():
        ret, frame = cap.read()

        frame = cv2.resize(frame, (480, 640))
        cv2.imshow('rtsp stream',frame)
        cv2.waitKey(1)

        k = cv2.waitKey(5) & 0xFF
        if k == ord('e'):
            cv2.destroyWindow('rtsp stream')
            break

    else:
        print("No Stream opened")

File: c++/tobii_example/tobii_example/eye_stream.cpp
#include "stdafx.h"
#include<opencv2/opencv.hpp>
#include <opencv/cv.h>
#include <iostream>
#include <string>

using namespace cv;
using namespace std;

int eye_stream_main()

{
	const std::string videoStreamAddress = "rtsp://192.168.71.50:8554/live/eyes";
	cv::VideoCapture capture(videoStreamAddress);

	if (!capture.open(videoStreamAddress)) {
		//Error
	}

	cv::namedWindow("eyes", CV_WINDOW_AUTOSIZE);

	cv::Mat frame;

	while (true) {
		if (!capture.read(frame)) {
			//Error
		} else { 
			cv::imshow("eyes", frame);

			cv::waitKey(30);
		}
	}
}

